src/command_modules/azure-cli-network/azure/cli/command_modules/network/_params.py

```diff
@@ -463,6 +463,14 @@ def load_arguments(self, _):
 
     # endregion
 
+    # region InterfaceEndpoint
+    interface_endpoint_name = CLIArgumentType(options_list='--endpoint-name', id_part='name', help='Name of the interface endpoint.', completer=get_resource_name_completion_list('Microsoft.Network/interfaceEndpoints'))
+
+    with self.argument_context('network interface-endpoint') as c:
+        c.argument('interface_endpoint_name', interface_endpoint_name, options_list=['--name', '-n'])
```

Comment:
Why does the `interface_endpoint_name` define a different options list if it is only being used here with a different options_list?

Reply:
Standard practice. Later, if there is a subresource `interface-endpoint foo` then I register it under that as just `c.argument('interface_endpoint_name`, interface_endpoint_name)`.

Reply:
I see, cool